backend/endpoints/volunteer.py

```diff
@@ -1,33 +1,77 @@
-from flask.views import MethodView
+import logging
+import os
+from datetime import datetime, timedelta
+from datetime import datetime as dt
+
 from flask import jsonify, request
+from flask.views import MethodView
+from passlib.utils import generate_password
 
+from config import MIN_PASSWORD_LEN, PassHash
 from endpoints.geo import calc_distance
-from models import Volunteer, Beneficiary, Operator
-from config import PassHash, MIN_PASSWORD_LEN
-from datetime import datetime as dt
-from datetime import datetime, timedelta
-import logging
+from models import Beneficiary, Operator, Volunteer
+from utils import volunteer_utils as vu
+
 log = logging.getLogger("back")
 
 
-def registerVolunteer(requestjson, created_by):
-        """create a new user"""
-        new_volunteer = requestjson
-        if len(created_by)>30:
+def register_volunteer(request_json, created_by):
+    """Creates and persists a new volunteer into database.
+
+    Parameters
+    ----------
+    request_json : dict
+        A dictionary representing the volunteer profile details.
+    created_by : str
+         A string representing either name of user who is going to create a new volunteer, or the token
+
+    Returns
+    -------
+
+    """
+    try:
+        if not vu.is_email(created_by):
             user = Operator.verify_auth_token(created_by)
             created_by = user.get().clean_data()['email']
-        # TODO: get authenticated operator and assignee to new volunteer
-        # new_volunteer["created_by"] = authenticated_oprator
-        try:
-            assert len(new_volunteer["password"]) >= MIN_PASSWORD_LEN, f"Password is to short, min length is {MIN_PASSWORD_LEN}"
-            new_volunteer["password"] = PassHash.hash(new_volunteer["password"])
-            assert not Volunteer.objects(email=new_volunteer['email']) , "user with this email already exists"
-            new_volunteer['created_by'] = created_by#g.user.get().clean_data()['_id']
-            comment = Volunteer(**new_volunteer)
-            comment.save()
-            return jsonify({"response": "success", 'user': comment.clean_data()})
-        except Exception as error:
-            return jsonify({"error": str(error)}), 400
+
+        if created_by == os.getenv("COVID_BACKEND_USER"):
```

Comment:
it limits the availability of the COVID_BACKEND_USER to add volunteers from UI, the tg-bot request can be identified using chat-id

Reply:
I think this is done.